api/src/config.rs

```diff
@@ -0,0 +1,81 @@
+//! All the configurations for the application
+//!
+//! Contains types definitions and traits implementations
+//! for managed states used with `.manage()` on a rocket app`
+
+use diesel::mysql::MysqlConnection;
+use r2d2;
+use r2d2_diesel::ConnectionManager;
+use rocket::http::RawStr;
+use rocket::http::Status;
+use rocket::request::{self, FromParam, FromRequest};
+use rocket::{Outcome, Request, State};
+use std::ops::Deref;
+
+/// Encapsulate a r2d2 Mysql connection pool.
+type Pool = r2d2::Pool<ConnectionManager<MysqlConnection>>;
+
+/// Connection request guard type: a wrapper around an r2d2 pooled connection.
+pub struct DbConn(pub r2d2::PooledConnection<ConnectionManager<MysqlConnection>>);
+
+/// A string representing admin token to access admin only endpoints
+#[derive(Clone, PartialEq)]
```

Comment:
https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-common-traits

Reply:
?

Reply:
All Raven's Rust project must follow the Rust API Guidelines, or at least most of it.
This has been discussed on Discord already (#rust)

The `C-COMMON-TRAITS` rule isn't followed as you are not deriving from all possible traits for type (like Debug).

Reply:
done for allowed traits. Other can't be implemented on r2d2